### PR TITLE
test(getAbsolutePath): updated for os specific

### LINKED ISSUE #874: build failing on windows
This was my fault - was too hasty to merge this PR:  https://github.com/sasjs/cli/pull/866

image

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Closes #874 

## Intent

Fix `OS` specific tests for `getAbsolutePath`, also fix any test fails on windows environment.

## Implementation

- Removed unnecessary `deleteFolder` statements as it throws error.
- Revisited and updated `db.spec` for failing test case.
- Used default timeout (`jest.server.config.js`) for:
  - `cbd.spec.server.js`
  - `run.spec.server.js`
- Updated tests `utils.spec.js`, having `path.sep` to check `OS` oriented paths and normalize paths.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [ ] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked (Powershell):
image
Server:
image
- [x] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

**Review by allanbowe — DISMISSED** [review]

File: src/commands/deploy/spec/cbd.spec.server.ts [hunk 1]
```diff
@@ -237,7 +237,7 @@ describe('sasjs cbd having stream app', () => {
 
       await updateLocalTarget(target.name, { appLoc })
     },
-    2 * 60 * 1000
+    5 * 60 * 1000
```
Comment (YuryShkoda) [inline src/commands/deploy/spec/cbd.spec.server.ts:240]: Why you've decided to increase the timeout?

File: src/commands/run/spec/run.spec.server.ts [hunk 2]
```diff
@@ -201,7 +201,7 @@ describe('sasjs run', () => {
         expect(result.log.includes(logPart)).toBeTruthy()
       },
 
-      60 * 1000
+      5 * 60 * 1000
```
Comment (YuryShkoda) [inline src/commands/run/spec/run.spec.server.ts:204]: Why you've decided to increase the timeout?

File: src/commands/db/spec/db.spec.ts [hunk 3]
```diff
@@ -20,15 +20,4 @@ describe(`sasjs db`, () => {
 
     await verifyStep('db')
   })
-
```
Comment (YuryShkoda) [inline src/commands/db/spec/db.spec.ts:23]: Why you've decided to delete this test case?

File: src/commands/compile/spec/compile.spec.ts [hunk 4]
```diff
@@ -453,7 +453,6 @@ describe('sasjs compile outside project', () => {
     })
 
     afterEach(async () => {
-      await deleteFolder(parentOutputFolder)
```
Comment (YuryShkoda) [inline src/commands/compile/spec/compile.spec.ts:456]: Why you've decided to remain `parentOutputFolder `?

File: src/utils/spec/utils.spec.ts [hunk 5]
```diff
@@ -166,7 +167,9 @@ describe('utils', () => {
       const relativePath = './my-path/xyz'
       const pathRelativeTo = '/home/abc'
 
-      const expectedAbsolutePath = '/home/abc/my-path/xyz'
+      const expectedAbsolutePath = isWindows()
```
Comment (YuryShkoda) [inline src/utils/spec/utils.spec.ts:170]: I suggest creating a variable `pathSep` and use it to join parts of the path.

**Review by YuryShkoda — CHANGES_REQUESTED** [review]
Please resolve all comments left.

File: src/commands/deploy/spec/cbd.spec.server.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/deploy/spec/cbd.spec.server.ts:240]: Last commit is actually about this, server is taking too long for these 2 tests locally and in [github ci](https://github.com/sasjs/cli/runs/3137496360)

You may have look at other current PR's [actions](https://github.com/sasjs/cli/pull/855/checks?check_run_id=3138966454)

File: src/commands/run/spec/run.spec.server.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/run/spec/run.spec.server.ts:204]: explained above.

File: src/commands/db/spec/db.spec.ts (hunk 3 above)
Comment (saadjutt01) [reply src/commands/db/spec/db.spec.ts:23]: I have looked into this test case, it's not doing anything. There's no purpose of this test case.

File: src/commands/compile/spec/compile.spec.ts (hunk 4 above)
Comment (saadjutt01) [reply src/commands/compile/spec/compile.spec.ts:456]: There's no usage of `parentOutputFolder` in this `describe` block

Comment (saadjutt01) [reply src/commands/compile/spec/compile.spec.ts:456]: And deleting such non-existing folder in windows, raise exception

File: src/utils/spec/utils.spec.ts (hunk 5 above)
Comment (saadjutt01) [reply src/utils/spec/utils.spec.ts]: Updated

File: src/commands/deploy/spec/cbd.spec.server.ts (hunk 1 above)
Comment (YuryShkoda) [reply src/commands/deploy/spec/cbd.spec.server.ts:240]: we can just use the default timeout set in `jest.server.config.js`

File: src/commands/db/spec/db.spec.ts (hunk 3 above)
Comment (YuryShkoda) [reply src/commands/db/spec/db.spec.ts:23]: it checks if `buildDB()` throws a proper error message if `db` folder does not exis.

File: src/commands/run/spec/run.spec.server.ts (hunk 2 above)
Comment (YuryShkoda) [reply src/commands/run/spec/run.spec.server.ts:204]: explained above

File: src/commands/deploy/spec/cbd.spec.server.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/deploy/spec/cbd.spec.server.ts:240]: updated

File: src/commands/run/spec/run.spec.server.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/run/spec/run.spec.server.ts:204]: fixed

File: src/commands/db/spec/db.spec.ts (hunk 3 above)
Comment (saadjutt01) [reply src/commands/db/spec/db.spec.ts:23]: I have re-written it correctly.

**Review by YuryShkoda — APPROVED** [review]

**Review by allanbowe — APPROVED** [review]

**Pull request merged**